crates/ott-balancer/src/discovery/dns.rs

```diff
@@ -0,0 +1,51 @@
+use async_trait::async_trait;
+use tracing::info;
+use trust_dns_resolver::TokioAsyncResolver;
+
+use super::*;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct DnsDiscoveryConfig {
+    /// The port that monoliths should be listening on for load balancer connections.
+    pub monolith_port: u16,
+    pub dns_server: Option<String>,
+    pub query: String
```

Comment:
Would be nice to have doc strings for these.
```suggestion
    /// The DNS server to query. Optional. If not provided, the system configuration will be used instead.
    pub dns_server: Option<String>,
    /// The A record to query. If using docker-compose, this should be the service name for the monolith.
    pub query: String
```